bindings/python/src/layers.rs

```diff
@@ -18,47 +18,26 @@
 use std::time::Duration;
 
 use ::opendal as od;
+use opendal::Operator;
 use pyo3::prelude::*;
 
-#[derive(FromPyObject)]
-pub enum Layer {
-    ConcurrentLimit(ConcurrentLimitLayer),
-    ImmutableIndex(ImmutableIndexLayer),
-    Retry(RetryLayer),
+pub trait PythonLayer: Send + Sync {
+    fn layer(&self, op: Operator) -> Operator;
 }
 
-#[pyclass(module = "opendal.layers")]
-#[derive(Clone)]
-pub struct ConcurrentLimitLayer(pub od::layers::ConcurrentLimitLayer);
+#[pyclass(module = "opendal.layers", subclass)]
+pub struct Layer(pub Box<dyn PythonLayer>);
```

Comment:
I can't see the point of putting a `Box<dyn PythonLayer>` here, could you elaborate?

Reply:
> I can't see the point of putting a `Box<dyn PythonLayer>` here, could you elaborate?

I want to avoid the pattern of enum upon layers.